lib/Log/Console.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2023 Robin Appelman <robin@icewind.nl>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\LogReader\Log;
+
+use OC\Log\LogDetails;
+use OC\SystemConfig;
+use OCA\LogReader\Command\Tail;
+
+/**
+ * Utility to write log messages to the console as they are emitted
+ */
+class Console extends LogDetails {
+	private int $level;
+	private int $terminalWidth;
+	private Formatter $formatter;
+
+	public function __construct(Formatter $formatter, SystemConfig $config, string $level, int $terminalWidth) {
+		parent::__construct($config);
+		$this->formatter = $formatter;
+		$this->level = self::parseLogLevel($level);
+		$this->terminalWidth = $terminalWidth;
+	}
+
+	public function log(int $level, string $app, array $entry) {
+		if ($level >= $this->level) {
+			$messageWidth = $this->terminalWidth - 8 - 18 - 6;
```

Comment:
magic numbers (and in following lines are more magic numbers)

Reply:
They are the column widths that "seem to work best"

Same number as https://github.com/nextcloud/logreader/blob/0718de62d4963d8ffe731cd7141db0b74e6503ce/lib/Command/Tail.php#L75

Reply:
Better kept in named constants imo.